### DISCUSSION TITLE
Bug: custom overlay won't close when pressing `esc` button or clicking outside (once having focus)

### DISCUSSION THREAD

**Discussion by yairEO** [gpbl/react-day-picker#1113] [Support]
Go here:
http://react-day-picker.js.org/examples/input-custom-overlay

1. (Mouse) click on the datepicker input, an "overlay" should show.
2. Click on any *white* area within the overlay, then press the <kbd>esc</kbd> key
3. Overlay should close, but nothing happens
4. Click anywhere *outside* of the overlay, and again the overlay should close but it's not.

This also happens when giving focus to the custom input within the overlay in the demo page.

**Comment by joafeldmann** [comment]
Further the overlay opens again / stays open when you focus the input, hit `esc` and then press/hold the `esc` key again
